MoltenVK/MoltenVK/Utility/MVKObjectPool.h

```diff
@@ -161,7 +161,7 @@ class MVKObjectPool : public MVKBaseObject {
      * contains no objects. This differs from the acquireObject() function, which creates
      * and return a new instance if this pool is empty. This method is not thread-safe.
      */
-    T* nextObject() {
+    virtual T* nextObject() {
```

Comment:
This function was deliberately not virtualized to allow the core operation of unlinking an object from the head of the list to be preferably inlined for performance efficiency. The polymorphic virtualization for custom types is established in `newObject()`, which is called with much less frequency (if the pool is serving its purpose).

Can we move the invocation of `setPurgeableState:` to `MVKMTLBufferAllocator::acquireMTLBufferRegion()`, after `MVKMTLBufferAllocationPool::acquireObject()`, instead of needing `MVKMTLBufferAllocationPool::nextObject()`?

Reply:
> This function was deliberately not virtualized to allow the core operation of unlinking an object from the head of the list to be preferably inlined for performance efficiency. The polymorphic virtualization for custom types is established in `newObject()`, which is called with much less frequency (if the pool is serving its purpose).

I was afraid of that.

> 
> Can we move the invocation of `setPurgeableState:` to `MVKMTLBufferAllocator::acquireMTLBufferRegion()`, after `MVKMTLBufferAllocationPool::acquireObject()`, instead of needing `MVKMTLBufferAllocationPool::nextObject()`?

Why didn't I think of that? Done.